crates/net/network/src/transactions/fetcher.rs

```diff
@@ -68,6 +77,40 @@ impl TransactionFetcher {
         self
     }
 
+    /// Drains any resolved [`GetPooledTransactions`] requests and queues them as [`FetchEvent`]s
+    /// in the [`TransactionFetcher`].
+    pub fn try_drain_inflight_requests(&mut self) {
+        self.advance_inflight_requests(DEFAULT_MAX_COUNT_CONCURRENT_REQUESTS)
+    }
+
+    /// Advances inflight requests by at most the given number of [`GetPooledTransactions`]
+    /// requests. The [`PooledTransactions`] responses are queued as [`FetchEvent`]s in the
+    /// [`TransactionFetcher`].
+    pub fn advance_inflight_requests(&mut self, steps: u32) {
+        // `FuturesUnordered` doesn't close when `None` is returned. so just return if empty.
+        // <https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=815be2b6c8003303757c3ced135f363e>
+        if self.inflight_requests.is_empty() {
+            return
+        }
+
+        let mut budget = steps;
+
+        loop {
```

Comment:
what's the purpose of the outer loop here?
this will result in a busy loop if the underlying stream reaches pending

Reply:
oh yeah, this is result of a clippy fix, clippy warns if a loop can be made a while loop